src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -583,26 +584,73 @@ export default class Pilot {
         }
 
         const airportModel = this._fms.arrivalAirportModel || this._fms.departureAirportModel;
-        const altitudeCheck = aircraftModel.validateNextAltitude(altitude, airportModel);
 
-        if (!altitudeCheck[0]) {
-            return altitudeCheck;
-        }
+        if ((speed == null) && altitude) {
+            const altitudeCheck = aircraftModel.validateNextAltitude(altitude, airportModel);
 
-        altitude = airportModel.clampWithinAssignableAltitudes(altitude);
+            if (!altitudeCheck[0]) {
+                return altitudeCheck;
+            }
 
-        const waypoint = this._fms.findWaypoint(fixName);
+            altitude = airportModel.clampWithinAssignableAltitudes(altitude);
 
-        waypoint.setAltitude(altitude);
-        this._mcp.setAltitudeFieldValue(altitude);
-        this._mcp.setAltitudeVnav();
+            const waypoint = this._fms.findWaypoint(fixName);
```

Comment:
If. `fixName` is constant within this method, I'd encourage you to lift this out of and above the conditional blocks so we only need to assign `waypoint` once within this method.

Reply:
Done 🎉